src/AppInstallerCLICore/Workflows/ImportExportFlow.cpp

```diff
@@ -0,0 +1,204 @@
+            if (sourceItr == requestsFromSource.end())
+            {
+                requestsFromSource.emplace_back(Utility::LocIndString{ sourceIdentifier }, packageVersion->GetSource()->GetDetails());
+                sourceItr = std::prev(requestsFromSource.end());
+            }
+
+            sourceItr->Packages.emplace_back(
+                packageVersion->GetProperty(PackageVersionProperty::Id),
+                packageVersion->GetProperty(PackageVersionProperty::Version).get(),
+                packageVersion->GetProperty(PackageVersionProperty::Channel).get());
+        }
+
+        context.Add<Execution::Data::PackageCollection>(std::move(versionsToExport));
+    }
+
+    void WriteImportFile(Execution::Context& context)
+    {
+        auto packages = PackagesJson::CreateJson(context.Get<Execution::Data::PackageCollection>());
+
+        std::filesystem::path outputFilePath{ context.Args.GetArg(Execution::Args::Type::OutputFile) };
+        std::ofstream outputFileStream{ outputFilePath };
+        outputFileStream << packages;
+    }
+
+    void ReadImportFile(Execution::Context& context)
+    {
+        std::filesystem::path importFilePath{ context.Args.GetArg(Execution::Args::Type::ImportFile) };
+
+        // TODO: Handle errors
+        Json::Value jsonRoot;
+        std::ifstream{ importFilePath } >> jsonRoot;
+
+        auto packages = PackagesJson::ParseJson(jsonRoot);
+        context.Add<Execution::Data::PackageCollection>(packages);
+
+        if (packages.RequestsFromSources.empty())
+        {
+            AICLI_LOG(CLI, Warning, << "No packages to install");
+            AICLI_TERMINATE_CONTEXT(APPINSTALLER_CLI_ERROR_NO_APPLICATIONS_FOUND);
+        }
+    }
+
+    void SearchPackagesForImport(Execution::Context& context)
+    {
+        auto availableSources = Repository::GetSources();
+        std::vector<std::shared_ptr<IPackageVersion>> packagesToInstall;
+
+        // Aggregated source with all the required sources.
+        // We keep it as the source of the root context to keep all the
+        // source objects alive for install.
+        auto aggregatedSource = std::make_shared<Repository::CompositeSource>("*ImportSource");
+
+        // Look for the packages needed from each source independently.
+        // If a package is available from multiple sources, this ensures we will get it from the right one.
+        for (auto& requiredSource : context.Get<Execution::Data::PackageCollection>().RequestsFromSources)
+        {
+            if (!requiredSource.Details.Name.empty())
+            {
+                // For packages that come from a specific source, find the matching available source.
+                // Match required sources with the available sources by their arguments, as they may have been added with a different name.
+                // TODO: Can we do this with identifiers without opening the source?
```

Comment:
Identifiers should be used, and they should not require the source to be opened to find them.

Reply:
Do you know if this is something we already have?
From what I found, the SQLIndexSource takes its identifier from the SourceDetails' Data (package family name) but we don't have anything in just the SourceDetails that gives the identifier directly, and that's all we have before opening the source. I didn't dig deep enough to find where the SourceDetails are first taken from to add the identifier there, or to know if it would be safe to always take the Data as the identifier.

Reply:
Added the Identifier to the SourceDetails and persisted it in the sources yaml file. When opening, the identifier is still calculated for the SQLIndexSource from the data instead of read from the file, so that it is available even if the file does not have it. For any future source type we can really read the value from the file. When adding a source, the identifier is also calculated the same way without needing to open the source.